server/eclipse-project/src/main/java/us/freeandfair/corla/asm/AsmTransitions.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Free & Fair Colorado RLA System
+ * 
+ * @title ColoradoRLA
+ * @created Aug 8, 2017
+ * @copyright 2017 Free & Fair
+ * @license GNU General Public License 3.0
+ * @author Joe Kiniry <kiniry@freeandfair.us>
+ * @description A system to assist in conducting statewide risk-limiting audits.
+ */
+
+package us.freeandfair.corla.asm;
+
+import us.freeandfair.corla.util.Pair;
+
+/**
+ * The generic idea of a set of ASM transitions.
+ * @trace asm.asm_transition_function
+ */
+public interface AsmTransitions {
+  /**
+   * The Department of State Dashboard's transition function.
+   * @trace asm.dos_dashboard_next_state
+   */
+  enum DosDashboardTransitions implements AsmTransitions {
+    AUTHENTICATE(
+      new Pair<AsmState, AsmEvent>(AsmState.DosDashboardState.INITIAL_STATE, 
+      AsmEvent.DosDashboardEvent.AUTHENTICATE_STATE_ADMINISTRATOR_EVENT),
+      AsmState.DosDashboardState.AUTHENTICATED),
+    SET_RISK_LIMIT(
+      new Pair<AsmState, AsmEvent>(AsmState.DosDashboardState.AUTHENTICATED, 
+      AsmEvent.DosDashboardEvent.ESTABLISH_RISK_LIMIT_FOR_COMPARISON_AUDITS_EVENT),
+      AsmState.DosDashboardState.RISK_LIMITS_SET),
+    CONTESTS_SELECTED(
+      new Pair<AsmState, AsmEvent>(AsmState.DosDashboardState.RISK_LIMITS_SET, 
+      AsmEvent.DosDashboardEvent.SELECT_CONTESTS_FOR_COMPARISON_AUDIT_EVENT),
+      AsmState.DosDashboardState.CONTESTS_TO_AUDIT_IDENTIFIED),
+    RANDOM_SEED(
+      new Pair<AsmState, AsmEvent>(AsmState.DosDashboardState.CONTESTS_TO_AUDIT_IDENTIFIED, 
+      AsmEvent.DosDashboardEvent.PUBLIC_SEED_EVENT),
+      AsmState.DosDashboardState.RANDOM_SEED_PUBLISHED),
+    PUBLISH_BALLOTS_TO_AUDIT(
+      new Pair<AsmState, AsmEvent>(AsmState.DosDashboardState.RANDOM_SEED_PUBLISHED, 
+      AsmEvent.DosDashboardEvent.PUBLISH_BALLOTS_TO_AUDIT_EVENT),
+      AsmState.DosDashboardState.BALLOT_ORDER_DEFINED),
```

Comment:
These are out of order. It is imperative that PUBLISH_BALLOTS_TO_AUDIT_EVENT comes before PUBLIC_SEED_EVENT